packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2409,21 +2409,16 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
   /// If spell check is enabled, this will try to infer a value for
   /// the [SpellCheckService] if left unspecified.
   static SpellCheckConfiguration _inferSpellCheckConfiguration(SpellCheckConfiguration? configuration) {
-    if (configuration == null || configuration == const SpellCheckConfiguration.disabled()) {
+    final SpellCheckService? spellCheckService = configuration?.spellCheckService;
+    final bool spellCheckServiceIsConfigured = spellCheckService != null || spellCheckService == null && WidgetsBinding.instance.platformDispatcher.nativeSpellCheckServiceDefined;
+    if (configuration == null || configuration == const SpellCheckConfiguration.disabled() || !spellCheckServiceIsConfigured) {
+      // Only enable spell check if a non-disabled configuration is provided
+      // and if that configuration does not specify a spell check service,
+      // a native spell checker must be supported.
       return const SpellCheckConfiguration.disabled();
     }
 
```

Comment:
Should we add a warning if `configuration != null && spellCheckService == null && !WidgetsBinding.instance.platformDispatcher.nativeSpellCheckServiceDefined`?  Something like:

"Spell check was enabled with spellCheckConfiguration, but the current platform does not have a supported spell check service, and none was provided."

Reply:
Here's something similar to what I'm thinking of:

https://github.com/flutter/flutter/blob/5b40de684a211d343e2ae901c9ea690bcb77b996/packages/flutter/lib/src/widgets/navigator.dart#L3527-L3541

This will not run in production mode since it's wrapped in an `assert`.  Also, FlutterError.reportError won't halt execution.